sdk/webpubsub/azure-messaging-webpubsubservice/azure/messaging/webpubsubservice/_patch.py

```diff
@@ -322,6 +273,86 @@ def from_connection_string(cls, connection_string, **kwargs):
         credential = AzureKeyCredential(kwargs.pop("accesskey"))
         return cls(credential=credential, **kwargs)
 
+    def _get_token_by_key(self, endpoint, hub, **kwargs):
+        # type: (str, str, Any) -> str
+        """build token with access key.
+
+        :param endpoint:  HTTPS endpoint for the WebPubSub service instance.
+        :type endpoint: str
+        :param hub: The hub to give access to.
+        :type hub: str
+        :returns: token
+        :rtype: str
+        """
+        audience = "{}/client/hubs/{}".format(endpoint, hub)
+        user = kwargs.pop("user_id", None)
+        key = self._config.credential.key
+        ttl = timedelta(minutes=kwargs.pop("expire_in_minutes", 60))
+        roles = kwargs.pop("roles", [])
+
+        payload = {
+            "aud": audience,
+            "iat": datetime.now(tz=_UTC_TZ()),
+            "exp": datetime.now(tz=_UTC_TZ()) + ttl,
+        }
+        if user:
+            payload["sub"] = user
+        if roles:
+            payload["role"] = roles
+
+        return six.ensure_str(jwt.encode(payload, key, algorithm="HS256"))
+
+    @distributed_trace
+    def get_client_access_token(self, hub, **kwargs):
+        # type: (str, Any) -> Dict[Any]
+        """Build an authentication token.
```

Comment:
This docstring should be updated to describe the hybrid nature of this method, and when it will build a token vs when it will call the service.